render-test/file_source.cpp

```diff
@@ -0,0 +1,69 @@
+#include <mbgl/storage/resource_options.hpp>
+
+#include "file_source.hpp"
+
+namespace mbgl {
+
+std::atomic_size_t requestCount{0};
+std::atomic_size_t transferredSize{0};
+std::atomic_bool active{false};
+
+ProxyFileSource::ProxyFileSource(const std::string& cachePath,
+                                 const std::string& assetPath,
+                                 bool supportCacheOnlyRequests_)
+    : DefaultFileSource(cachePath, assetPath, supportCacheOnlyRequests_) {}
+
+ProxyFileSource::ProxyFileSource(const std::string& cachePath,
+                                 std::unique_ptr<FileSource>&& assetFileSource_,
+                                 bool supportCacheOnlyRequests_)
+    : DefaultFileSource(cachePath, std::move(assetFileSource_), supportCacheOnlyRequests_) {}
+
+ProxyFileSource::~ProxyFileSource() = default;
+
+std::unique_ptr<AsyncRequest> ProxyFileSource::request(const Resource& resource, Callback callback) {
+    auto result = DefaultFileSource::request(resource, [=](Response response) {
+        std::size_t size = response.data != nullptr ? response.data->size() : 0;
+        if (active) {
+            requestCount++;
+            transferredSize += size;
+        }
+        callback(response);
+    });
+    return result;
+}
+
+std::shared_ptr<FileSource> FileSource::createPlatformFileSource(const ResourceOptions& options) {
+    auto fileSource = std::make_shared<ProxyFileSource>(
+        options.cachePath(), options.assetPath(), options.supportsCacheOnlyRequests());
+    fileSource->setAccessToken(options.accessToken());
+    fileSource->setAPIBaseURL(options.baseURL());
+    return fileSource;
+}
+
+// static
+void ProxyFileSource::setTrackingActive(bool active_) {
+    active = active_;
+}
+
+// static
+bool ProxyFileSource::isTrackingActive() {
+    return active;
+}
+
+// static
+void ProxyFileSource::resetMetrics() {
```

Comment:
This method is not needed, could be implicit inside `::setTrackingActive` if you change the state of `active`.

Reply:
Done